File: mustachio.frontend/src/app/dto/reservation-editor-dto.ts
import { Reservation } from "../model/reservation";

export class ReservationEditorDto {
  userId: string;
  startsAt: Date;
  endsAt: Date;

  constructor(userId: string, startsAt: Date, endsAt: Date | null = null) {
    this.userId = userId;
    this.startsAt = startsAt;
    this.endsAt = endsAt ?? new Date(new Date(startsAt).setHours(startsAt.getHours() + Reservation.defaultDurationHours));
  }
}

File: mustachio.api/src/main/java/pg/decola_tech_avanade_2025/desafios/mustachio/api/dto/ReservationEditorDto.java
package pg.decola_tech_avanade_2025.desafios.mustachio.api.dto;

import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.beans.BeanUtils;
import pg.decola_tech_avanade_2025.desafios.mustachio.api.model.Reservation;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class ReservationEditorDto {
    @NotNull(message = "Property must not be null or empty.")
    private UUID userId;

    @NotNull(message = "Property must not be null or empty.")
    private LocalDateTime startsAt;

    private LocalDateTime endsAt;

    public static ReservationEditorDto fromReservation(Reservation model) {
        ReservationEditorDto instance = new ReservationEditorDto();
        BeanUtils.copyProperties(model, instance);
        return instance;
    }
}
